File: 240124/작은 구슬의 이동/small-marble-movement.py
def in_range(x, y):
    return 1 <= x <= n and 1 <= y <= n 

n, t = map(int, input().split())
x, y, dir = input().split()
x, y = int(x), int(y)
arr = [[0 for j in range(n)] for i in range(n)]

dx = [1, 0, 0, -1]
dy = [0, -1, 1, 0]

d = {
    'D': 0,
    'L': 1,
    'R': 2,
    'U': 3
}

dir_num = d[dir]

for _ in range(t):
    nx = x + dx[dir_num]
    ny = y + dy[dir_num]

    if in_range(nx, ny):
        x, y = nx, ny
    else:
        dir_num = 3 - dir_num

print(x, y)

File: 240911/작은 구슬의 이동/small-marble-movement.js
const fs = require("fs");
let input = fs.readFileSync(0).toString().trim().split("\n");

let [n, t] = input[0].trim().split(" ").map(Number);
let [r, c, d] = input[1].trim().split(" ");
r = Number(r); c = Number(c); 

let mapper = {
    "R": 0,
    "U": 1,
    "L": 2,
    "D": 3
};

let dx = [0, -1, 0, 1], dy = [1, 0, -1, 0];

let cDir = mapper[d];

function inRange(x, y) {
    return x > 0 && x <= n && y > 0 && y <= n;
}

for (let i = 0; i < t; i++) {
    let nx = r + dx[cDir], ny = c + dy[cDir];

    if (!inRange(nx, ny)) {
        cDir = (cDir + 2) % 4;
        continue;
    }

    r = r + dx[cDir]; c = c + dy[cDir];    
    // console.log(r, c);
}

console.log(r, c);